sway/commands/output.c

```diff
@@ -260,40 +30,31 @@ struct cmd_results *cmd_output(int argc, char **argv) {
 		wlr_log(L_ERROR, "Failed to allocate output config");
 		return NULL;
 	}
+	argc--; argv++;
 
-	for (int i = 1; i < argc; ++i) {
-		const char *command = argv[i];
+	config->handler_context.output_config = output;
 
-		if (strcasecmp(command, "enable") == 0) {
-			output->enabled = 1;
-		} else if (strcasecmp(command, "disable") == 0) {
-			output->enabled = 0;
-		} else if (strcasecmp(command, "mode") == 0 ||
-				strcasecmp(command, "resolution") == 0 ||
-				strcasecmp(command, "res") == 0) {
-			error = cmd_output_mode(output, &i, argc, argv);
-		} else if (strcasecmp(command, "position") == 0 ||
-				strcasecmp(command, "pos") == 0) {
-			error = cmd_output_position(output, &i, argc, argv);
-		} else if (strcasecmp(command, "scale") == 0) {
-			error = cmd_output_scale(output, &i, argc, argv);
-		} else if (strcasecmp(command, "transform") == 0) {
-			error = cmd_output_transform(output, &i, argc, argv);
-		} else if (strcasecmp(command, "background") == 0 ||
-				strcasecmp(command, "bg") == 0) {
-			error = cmd_output_background(output, &i, argc, argv);
-		} else if (strcasecmp(command, "dpms") == 0) {
-			error = cmd_output_dpms(output, &i, argc, argv);
+	while (argc > 0) {
+		if (find_handler(*argv, output_handlers, sizeof(output_handlers))) {
+			error = config_subcommand(argv, argc, output_handlers,
+					sizeof(output_handlers));
 		} else {
 			error = cmd_results_new(CMD_INVALID, "output",
-				"Invalid output subcommand: %s.", command);
+				"Invalid output subcommand: %s.", *argv);
 		}
 
 		if (error != NULL) {
 			goto fail;
 		}
+
+		argc = config->handler_context.leftovers.argc;
+		argv = config->handler_context.leftovers.argv;
```

Comment:
If the handler "forgets" to set `leftovers` in some cases, this will become an infinite loop. Is it possible to reset them to NULL before handling and assert they're not NULL after handling?